scripts/locale-checks.ts

```diff
@@ -0,0 +1,318 @@
+      }
+      return acc;
+    },
+    {},
+  );
+
+  // Now, retrieve martyrology keys evaluated from inputs, to combine them with the ones from definitions.
+  // This is necessary because the inputs may have dropped some martyrology keys (when a martyrology key is not found).
+  const martyrologyComputedKeys = Object.values(inputs).reduce<Set<string>>((acc, value) => {
+    const mkeys = martyrologyComputedKeysFromDefinitions[value.id]
+      ? martyrologyComputedKeysFromDefinitions[value.id]
+      : value.martyrology?.map((m) => (typeof m === 'string' ? m : m.id));
+    mkeys.forEach((mkey) => acc.add(mkey));
+
+    return acc;
+  }, new Set<string>());
+
+  // Retrieve locale computed keys from inputs
+  const localeComputedKeys = Object.values(inputs).reduce<string[]>(
+    (acc, value) => [...acc, value.i18nDef[0]],
+    metaI18nKeys,
+  );
+
+  allCalendarData[calendar.name] = {
+    localeComputedKeys,
+    martyrologyComputedKeys: Array.from(martyrologyComputedKeys),
+  };
+
+  localeComputedKeys.forEach((key) => allLocalesKeys.add(key));
+  martyrologyComputedKeys.forEach((key) => allMartyrologyKeys.add(key));
+}
+
+/**
+ * [1] If there are useless martyrology items.
+ */
+const uselessMartyrologyKeys = findMissingInArray(Object.keys(Martyrology.catalog), allMartyrologyKeys);
+logIf('error', `Useless martyrology items:`, uselessMartyrologyKeys);
+
+/**
+ * [2] If there are missing martyrology items.
+ */
+const missingMartyrologyKeys = findMissingInArray(allMartyrologyKeys, Object.keys(Martyrology.catalog));
+logIf('error', `Missing martyrology items:`, missingMartyrologyKeys);
+
+/**
+ * [3] If there are missing localized 'en' items
+ */
+const missingEnLocaleKeys = findMissingLocalizedItems('En', allLocalesKeys);
+logIf('error', `Missing localized '${u('en')}' items:`, missingEnLocaleKeys);
+
+Object.keys(Martyrology.catalog).forEach((key) => allLocalesKeys.add(`names:${key}`));
+
+/**
+ * [4] If there are useless localized name items.
+ */
+Object.keys(locales).forEach((localeKey) => {
+  const uselessLocalizedKeys = findMissingLocalizedItems(localeKey, allLocalesKeys, {
+    reverse: true,
+    localeKeyNamesOnly: true,
+  });
+  logIf('error', `Useless localized '${toPackageName(localeKey)}' items:`, uselessLocalizedKeys);
+});
+
+/**
+ * [5] If the martyrology items are not sorted alphabetically.
+ */
+const areNotSortedMartyrologyKeys =
+  Object.keys(Martyrology.catalog).sort().join('') !== Object.keys(Martyrology.catalog).join('');
+logIf('error', `Martyrology keys are not sorted alphabetically.`, areNotSortedMartyrologyKeys);
+
+/**
+ * [6] If the localized name items are not sorted alphabetically.
+ */
+Object.values(locales).forEach((locale) => {
+  const areNotSortedNames =
+    Object.keys(locale.names ?? {})
+      .sort()
+      .join('') !== Object.keys(locale.names ?? {}).join('');
+  logIf('error', `Localized '${u(locale.id)}' names are not sorted alphabetically.`, areNotSortedNames);
+});
```

Comment:
Didn’t you want to create a function to check whether the keys are sorted? You use nearly the same code, apart from the input object (`Martyrology.catalog` vs `locale.names ?? {}`) and the error string (`Martyrology keys are not sorted alphabetically.` vs `Localized '${u(locale.id)}' names are not sorted alphabetically.`).